### ISSUE TITLE
Platform-dependent paths in extra CSS

### ISSUE THREAD

**Issue by orlade** [mkdocs/mkdocs#306]
On Windows, if I create some file `/css/foo.css`, it is included in the final served HTML document as `<link href="../css\override.css" rel="stylesheet">`. Firefox returns a 404 on the file and doesn't load the styles. Chrome and IE11 rewrite the URL as `../css/foo.css` when they send the request, so the styles are loaded.

CSS (and JS) files included in the rendered page should only use forward slashes, regardless of the platform they are built on.

**Comment by Lunatrius** [comment]
Using the master branch:
```html
  <link rel="stylesheet" href="../css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../css/theme_extra.css" type="text/css" />
  <link rel="stylesheet" href="../css/highlight.css">
  <link href="../css\test.css" rel="stylesheet">
```
```log
"GET /css/theme.css HTTP/1.1" 200 -
"GET /css/theme_extra.css HTTP/1.1" 200 -
"GET /css/highlight.css HTTP/1.1" 200 -
code 404, message File not found
"GET /css%5Ctest.css HTTP/1.1" 404 -
```

**Comment by d0ugal** [comment]
Thanks for confirming @Lunatrius. I'll try and have a look at this, I don't have a Windows machine I can easily test on however which makes things a bit trickier.

**Comment by d0ugal** [comment]
We currently have tests failing on Windows - I suspect the issue is related. https://github.com/mkdocs/mkdocs/pull/396 (see the appveyor CI failures)

**Comment by Lunatrius** [comment]
I usually use this *hack* when saving/writing relative paths (there is most likely a better alternative to this):
```py
__REPLACE__ = os.path.sep != '/'

def relpath(path, relroot=None):
    if relroot:
        path = os.path.relpath(path, relroot)

    if __REPLACE__:
        path = path.replace(os.path.sep, '/')

    return path
```

**Comment by d0ugal** [comment]
Interesting, that could work well.

**Comment by d0ugal** [comment]
@orlade, @Lunatrius Would either (or both!) of you be able to try out #451? It should fix your issues, I've only been able to do limited Windows testing. It's a fairly large refactor, so hoping to get as many people look at it as possible.

**Comment by Lunatrius** [comment]
That does fix it.

**Comment by d0ugal** [comment]
Thanks @Lunatrius

**Comment by d0ugal** [comment]
Fixed in #451
